src/main/java/com/clouway/cuse/spi/SearchEngineImpl.java

```diff
@@ -38,6 +38,16 @@ public void register(Object instance) {
 
   }
 
+  public void registerAll(List<Object> objects){
+    Class instanceClass=objects.get(0).getClass();
```

Comment:
hm, maybe we could check the list  - null and size, I think the mathid invocation should be save and to ignore cases where no objects are passed, 
Other thing that we could check(maybe in another PR) is the existence of a @SearchIndex annotaion - which deifnes that the object is a search index - this will prevent the new users from trying to store not valid index objects